bpf/process/types/basic.h

```diff
@@ -1536,13 +1538,26 @@ static inline __attribute__((always_inline)) int match_binaries(__u32 selidx)
 			if (!path_map)
 				return 0;
 			found_key = map_lookup_elem(path_map, current->bin.path);
-			match = !!found_key;
+			break;
+		case op_filter_prefix:
+		case op_filter_notprefix:
+			path_map = map_lookup_elem(&string_prefix_maps, &selector_options->map_id);
+			if (!path_map)
+				return 0;
+			// prepare the key on the stack to perform lookup in the LPM_TRIE
+			memset(&prefix_key, 0, sizeof(prefix_key));
+			prefix_key.prefixlen = current->bin.path_length * 8; // prefixlen is in bits
+			copy_length = probe_read_str(prefix_key.data, current->bin.path_length & (STRING_PREFIX_MAX_LENGTH - 1), current->bin.path);
```

Comment:
Should this not be a probe_read instead of probe_read_str? The latter will include a \0 at the end of the string, included in the length.

Reply:
And actually, maybe a path can contain a \0 at which point the probe_read_str would stop copying?

Reply:
I'm not sure a unix path can contain a `\0` but indeed we don't mind using `probe_read` or `probe_read_str` here. `probe_read_str` will return the copied_length while `probe_read` will return 0 on success, anyway here we just check that the return value is `< 0`. In the end it can copy a max of `256 - 1` in a buffer of size `256` with all values set to 0 so it won't change much, it seems that `prefix_key.data[255]` will always be `0` anyway.

I guess it's a cheaper approach to use `probe_rea`d since we know the size in advance?